README.md

```diff
@@ -125,15 +125,16 @@ so I suggest you define them in `~/.config/nvim/after/ftplugin/rust.lua`[^1]
 Example:
 
 ```lua
-local bufnr = vim.api.nvim_get_current_buf()
-vim.keymap.set(
-  "n", 
-  "<leader>a", 
-  function()
-    vim.cmd.RustLsp('codeAction') 
-  end,
-  { silent = true, buffer = bufnr }
-)
+vim.g.rustaceanvim = {
+  server = {
+    on_attach = function(client, bufnr)
+      -- semantic highlighting is poorly supported by many themes. It also causes the colors to change after rust-analyzer ran.
+      client.server_capabilities.semanticTokensProvider = nil
```

Comment:
question: What's the behaviour when using a theme that doesn't support semantic highlighting? Does it behave in an undesirable way? If so, I would suggest opening an issue with Neovim, because that isn't specific to this plugin.

For themes that support semantic highlighting, I don't think colours changing after the LSP client is attached is necessarily a bad thing.

Reply:
I don't remember the exact issue, but I think it was something that the highlight groups change in an incompatible way when they are coming from rust-analyzer. I tried multiple themes, which were supposed to be compatible and encountered the same issue with all.

It's seems to be a pretty common issue people run into, I've seen it several times on reddit.

![Screenshot_before](https://github.com/mrcjkb/rustaceanvim/assets/1109503/a18f8977-8148-4b22-a482-1bcaef1438d1)
![Screenshot_after](https://github.com/mrcjkb/rustaceanvim/assets/1109503/4d836573-5b53-4a25-b9b2-168858de04f2)

Reply:
If that's the case, I would suggest identifying the highlight group with `:Inspect` and either

- Requesting or PRing support to the colortheme you're using.
- Adding the missing highlight groups to your personal config.

For example

```lua
vim.api.nvim_set_hl(0, "@lsp.typemod.variable.mutable", { link = "@variable" })
```

:thinking: Perhaps it's a good idea to identify the problematic highlight groups and add a command to this plugin that creates appropriate links, e.g. `:RustLsp fixHighlights`.

![scrot](https://github.com/mrcjkb/rustaceanvim/assets/12857160/38e1757a-ff0a-4fd9-8404-d69e96262f51)

Reply:
With `:Inspect` I don't see a difference, I think it may be related to the tags or something like this. Sorry, but I don't have time to investigate more on a feature I don't want. Highlighting works flawless for me without LSP.

I think information for this setting is useful for others too.

Reply:
You are proposing to recommend to disable a feature that may be useful (or not disruptive) for most people in the quick setup section of this project's readme.

I really don't see that as necessary. If enough people complain about it, I will look into providing a proper fix.